### DISCUSSION TITLE
Can't add gas source

### DISCUSSION THREAD

**Discussion by kosymodo** [BottlecapDave/HomeAssistant-OctopusEnergy#244] [Q&A]
Hey folks - I've been setting this up from scratch today.  I have managed to set up the electricity source very easily, but can't set up the gas source.  The only entity/sensor available named gas is the previous accumulative consumption.  Not sure why I can't see the current consumption entity/sensor?  Any ideas?

**Comment by BottlecapDave** [comment]
Hi there. Which sensors are you missing? Is this with a Home Mini? If so, do you see gas available in the app?

**Reply by kosymodo** [reply]
The missing sensor is the **sensor.octopus_energy_gas_{{METER_SERIAL_NUMBER}}_{{MPRN_NUMBER}}_current_consumption** sensor, according to your instructions

Yeah it's with a Home Mini

Are you referring to the Octopus app?  If so, under Gas, it says "Smart gas meters only send readings every 30 minutes. Tap 'Day' for a grpah view of today's gas usage" i.e. I can only see live Electricity usage

**Reply by BottlecapDave** [reply]
Unfortunately the integration only retrieves the same data that the Octopus Energy app exposes. If you can't see the gas meter under "today" for "day" then it won't be available in the integration. However I'm sure when I first setup the Octopus Mini it took a few days for gas to start appearing. When it appears in the app, if you reload the integration (or restart your instance of HA) then it should appear.

I'll update the docs to reflect this situation.

**Reply by kosymodo** [reply]
Thanks Dave - what I don't understand though is why you'd be able to view live gas consumption through your smart meter, but I can't through mine?  The Octopus message states that no smart gas meter sends live readings?

BTW - I can see the gas meter under "today" for "day", but not under "live"

**Reply by BottlecapDave** [reply]
Wires might be crossed. I only see gas appear in the "day" section. I don't have it under "live", but the data filters in during the day (which is better than having to wait till the next day. Could you follow the instructions in the [faq](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/_docs/faq.md#ive-been-asked-for-my-meter-information-in-a-bug-request-how-do-i-obtain-this) and paste the contents here. This would give a clue to why the gas sensor is not appearing.

**Reply by kosymodo** [reply]
Ah right! ;)

{
  "home_assistant": {
    "installation_type": "Home Assistant Container",
    "version": "2022.11.4",
    "dev": false,
    "hassio": false,
    "virtualenv": false,
    "python_version": "3.10.7",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/London",
    "os_name": "Linux",
    "os_version": "5.15.0-72-generic",
    "run_as_root": true
  },
  "custom_components": {
    "hacs": {
      "version": "1.32.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "octopus_energy": {
      "version": "7.1.3",
      "requirements": []
    }
  },
  "integration_manifest": {
    "domain": "octopus_energy",
    "name": "Octopus Energy",
    "codeowners": [
      "@bottlecapdave"
    ],
    "config_flow": true,
    "dependencies": [
      "repairs",
      "recorder"
    ],
    "documentation": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/",
    "homekit": {},
    "iot_class": "cloud_polling",
    "issue_tracker": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues",
    "ssdp": [],
    "version": "7.1.3",
    "zeroconf": [],
    "is_built_in": false
  },
  "data": {
    "electricity_meter_points": [
      {
        "mpan": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "is_export": false,
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1063 - Landis and Gyr",
            "model": "00050202",
            "firmware": "38040404"
          }
        ],
        "agreements": [
          {
            "valid_from": "2023-04-15T23:00:00+00:00",
            "valid_to": null,
            "tariff_code": "E-1R-VAR-22-10-01-H",
            "product_code": "VAR-22-10-01"
          }
        ]
      }
    ],
    "gas_meter_points": [
      {
        "mprn": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "consumption_units": "m\u00b3",
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1063 - Landis and Gyr",
            "model": "47720101",
            "firmware": "03033254"
          }
        ],
        "agreements": [
          {
            "valid_from": "2022-08-03T23:00:00+00:00",
            "valid_to": "2023-08-03T23:00:00+00:00",
            "tariff_code": "G-1R-SILVER-2017-1-H",
            "product_code": "SILVER-2017-1"
          }
        ]
      }
    ]
  }
}

**Reply by BottlecapDave** [reply]
That output suggests that you should have a gas consumption sensor available. Have you tried reloading the integration or restarting your HA instance? Do you have any errors in your logs?

**Reply by kosymodo** [reply]
Yeah I've tried reloading the integration (just did it again now) and restarting HA (again, just did it now again), but no luck.

Can't see any errors in the logs.

It's definitely there in the list of entities (Settings > Devices & Services > Entities), but can't be selected (i.e. doesn't appear) in the 'Configure Gas Consumption' dropdown when in the energy config screen

**Reply by BottlecapDave** [reply]
Sorry more wires crossed. I thought the entity wasn't available at all, not that it was not available within the energy dashboard.

I've just removed my integration and reset my database and was still able to add the current consumption for gas on my instance. I know this is annoying, but could you go into the "Configure gas consumption" view, search for gas and take a screenshot and post it here?

**Reply by kosymodo** [reply]
Sorry for the delay Dave - work got in the way!

I have now reinstalled HA completely, and I now seem to have everything set up and working (albeit that I've now got a 'statistics_not_defined' error on the Gas consumption for the current_consumption sensor, and a 'Entity unavailable' error on a current_consumption (unknown) sensor.  Perhaps they'll resolve in due course...

**Reply by BottlecapDave** [reply]
hmm that doesn't sound right. Have you got any errors in your logs?

**Reply by kosymodo** [reply]
By way of an update Dave - after I set it up again last week, the sensors did update themselves at some point, and all has been working faultlessly since then :)
